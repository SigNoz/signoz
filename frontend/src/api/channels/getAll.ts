import axios from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { Channels, PayloadProps } from 'types/api/channels/getAll';

const getAll = async (): Promise<SuccessResponseV2<Channels[]>> => {
	try {
		const response = await axios.get<PayloadProps>('/channels');

		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
		throw error;
	}
};

export default getAll;
