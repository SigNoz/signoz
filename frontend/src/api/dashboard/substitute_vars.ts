import { ApiV5Instance } from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { QueryRangePayloadV5 } from 'api/v5/v5';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { ICompositeMetricQuery } from 'types/api/alerts/compositeQuery';

interface ISubstituteVars {
	compositeQuery: ICompositeMetricQuery;
}

export const getSubstituteVars = async (
	props?: Partial<QueryRangePayloadV5>,
	signal?: AbortSignal,
	headers?: Record<string, string>,
): Promise<SuccessResponseV2<ISubstituteVars>> => {
	try {
		const response = await ApiV5Instance.post<{ data: ISubstituteVars }>(
			'/substitute_vars',
			props,
			{
				signal,
				headers,
			},
		);

		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
	}
};
