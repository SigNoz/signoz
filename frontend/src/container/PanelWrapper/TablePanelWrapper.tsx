import { PANEL_TYPES } from 'constants/queryBuilder';
import GridTableComponent from 'container/GridTableComponent';
import { GRID_TABLE_CONFIG } from 'container/GridTableComponent/config';

import { PanelWrapperProps } from './panelWrapper.types';

function TablePanelWrapper({
	widget,
	queryResponse,
	tableProcessedDataRef,
	searchTerm,
}: PanelWrapperProps): JSX.Element {
	const panelData =
		(queryResponse.data?.payload?.data?.result?.[0] as any)?.table || [];
	const { thresholds } = widget;
	return (
		<GridTableComponent
			data={panelData}
			query={widget.query}
			thresholds={thresholds}
			columnUnits={widget.columnUnits}
			tableProcessedDataRef={tableProcessedDataRef}
			sticky={widget.panelTypes === PANEL_TYPES.TABLE}
			searchTerm={searchTerm}
			// eslint-disable-next-line react/jsx-props-no-spreading
			{...GRID_TABLE_CONFIG}
		/>
	);
}

export default TablePanelWrapper;
