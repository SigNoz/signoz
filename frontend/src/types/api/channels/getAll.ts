export type PayloadProps = {
	data: Channels[];
	status: string;
};

export interface Channels {
	created_at: string;
	data: string;
	id: string;
	name: string;
	type: string;
	updated_at: string;
}
