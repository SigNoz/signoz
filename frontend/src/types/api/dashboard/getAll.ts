import { PANEL_GROUP_TYPES, PANEL_TYPES } from 'constants/queryBuilder';
import { ThresholdProps } from 'container/NewWidget/RightContainer/Threshold/types';
import { timePreferenceType } from 'container/NewWidget/RightContainer/timeItems';
import { ReactNode } from 'react';
import { Layout } from 'react-grid-layout';
import { Query } from 'types/api/queryBuilder/queryBuilderData';

import { IField } from '../logs/fields';
import { BaseAutocompleteData } from '../queryBuilder/queryAutocompleteResponse';

export type PayloadProps = Dashboard[];

export const VariableQueryTypeArr = ['QUERY', 'TEXTBOX', 'CUSTOM'] as const;
export type TVariableQueryType = typeof VariableQueryTypeArr[number];

export const VariableSortTypeArr = ['DISABLED', 'ASC', 'DESC'] as const;
export type TSortVariableValuesType = typeof VariableSortTypeArr[number];

export interface IDashboardVariable {
	id: string;
	order?: any;
	name?: string; // key will be the source of truth
	description: string;
	type: TVariableQueryType;
	// Query
	queryValue?: string;
	// Custom
	customValue?: string;
	// Textbox
	textboxValue?: string;

	sort: TSortVariableValuesType;
	multiSelect: boolean;
	showALLOption: boolean;
	selectedValue?:
		| null
		| string
		| number
		| boolean
		| (string | number | boolean)[];
	// Internal use
	modificationUUID?: string;
	allSelected?: boolean;
	change?: boolean;
}
export interface Dashboard {
	id: number;
	uuid: string;
	createdAt: string;
	updatedAt: string;
	createdBy: string;
	updatedBy: string;
	data: DashboardData;
	isLocked?: boolean;
}

export interface DashboardTemplate {
	name: string;
	icon: React.ReactElement;
	id: string;
	description: string;
	previewImage: string;
}

export interface DashboardData {
	uuid?: string;
	description?: string;
	tags?: string[];
	name?: string;
	widgets?: Array<WidgetRow | Widgets>;
	title: string;
	layout?: Layout[];
	panelMap?: Record<string, { widgets: Layout[]; collapsed: boolean }>;
	variables: Record<string, IDashboardVariable>;
	version?: string;
	image?: string;
}

export interface WidgetRow {
	id: string;
	panelTypes: PANEL_GROUP_TYPES;
	title: ReactNode;
	description: string;
}

export interface ColumnUnit {
	[key: string]: string;
}
export interface IBaseWidget {
	isStacked: boolean;
	id: string;
	panelTypes: PANEL_TYPES;
	title: ReactNode;
	description: string;
	opacity: string;
	nullZeroValues: string;
	timePreferance: timePreferenceType;
	stepSize?: number;
	yAxisUnit?: string;
	stackedBarChart?: boolean;
	bucketCount?: number;
	bucketWidth?: number;
	mergeAllActiveQueries?: boolean;
	thresholds?: ThresholdProps[];
	softMin: number | null;
	softMax: number | null;
	fillSpans?: boolean;
	columnUnits?: ColumnUnit;
	selectedLogFields: IField[] | null;
	selectedTracesFields: BaseAutocompleteData[] | null;
}
export interface Widgets extends IBaseWidget {
	query: Query;
}

export interface PromQLWidgets extends IBaseWidget {
	query: { query: string; legend: string }[];
}

export interface IQueryBuilderTagFilterItems {
	id: string;
	key: string;
	op: string;
	value: string[];
}
