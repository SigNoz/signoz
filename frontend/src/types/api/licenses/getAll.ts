import { License } from './def';

export type PayloadProps = {
	licenses: License[];
};
