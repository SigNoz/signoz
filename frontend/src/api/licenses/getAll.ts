import { ApiV2Instance as axios } from 'api';
import { ErrorResponse, SuccessResponse } from 'types/api';
import { PayloadProps } from 'types/api/licenses/getAll';

const getAll = async (): Promise<
	SuccessResponse<PayloadProps> | ErrorResponse
> => {
	const response = await axios.get('/licenses');

	return {
		statusCode: 200,
		error: null,
		message: response.data.status,
		payload: response.data.data,
	};
};

export default getAll;
