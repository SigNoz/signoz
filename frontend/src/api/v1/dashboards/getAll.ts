import axios from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { Dashboard, PayloadProps } from 'types/api/dashboard/getAll';

const getAll = async (): Promise<SuccessResponseV2<Dashboard[]>> => {
	try {
		const response = await axios.get<PayloadProps>('/dashboards');
		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
	}
};

export default getAll;
