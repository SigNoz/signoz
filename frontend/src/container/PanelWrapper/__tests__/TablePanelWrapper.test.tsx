import { render } from 'tests/test-utils';
import { Widgets } from 'types/api/dashboard/getAll';

import TablePanelWrapper from '../TablePanelWrapper';
import {
	tablePanelQueryResponse,
	tablePanelWidgetQuery,
} from './tablePanelWrapperHelper';

describe('Table panel wrappper tests', () => {
	it('table should render fine with the query response and column units', () => {
		const { container, getByText } = render(
			<TablePanelWrapper
				widget={(tablePanelWidgetQuery as unknown) as Widgets}
				queryResponse={(tablePanelQueryResponse as unknown) as any}
				onDragSelect={(): void => {}}
			/>,
		);
		// checking the overall rendering of the table
		expect(container).toMatchSnapshot();

		// the first row of the table should have the latency value with units
		expect(getByText('4.35 s')).toBeInTheDocument();

		// the rows should have optimised value for human readability
		expect(getByText('31.3 ms')).toBeInTheDocument();

		// the applied legend should appear as the column header
		expect(getByText('latency-per-service')).toBeInTheDocument();
	});
});
