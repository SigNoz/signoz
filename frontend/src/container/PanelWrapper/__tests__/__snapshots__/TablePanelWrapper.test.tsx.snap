// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table panel wrappper tests table should render fine with the query response and column units 1`] = `
.c1 {
  position: absolute;
  right: -0.313rem;
  bottom: 0;
  z-index: 1;
  width: 0.625rem;
  height: 100%;
  cursor: col-resize;
}

.c0 {
  height: 95%;
  overflow: hidden;
}

.c0 .ant-table-wrapper {
  height: 100%;
}

.c0 .ant-spin-nested-loading {
  height: 100%;
}

.c0 .ant-spin-container {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .ant-table {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c0 .ant-table > .ant-table-container > .ant-table-content > table {
  min-width: 99% !important;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="query-table"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-2i2tap"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-2i2tap"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-small ant-table-layout-fixed ant-table-scroll-horizontal"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-header ant-table-sticky-holder"
                  style="overflow: hidden; top: 0px;"
                >
                  <table
                    style="table-layout: fixed; visibility: hidden;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          aria-label="service_name"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              service_name
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                aria-hidden="true"
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                          <span
                            class="c1 react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="latency-per-service"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              latency-per-service
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                aria-hidden="true"
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                          <span
                            class="c1 react-resizable-handle"
                          />
                        </th>
                      </tr>
                    </thead>
                  </table>
                </div>
                <div
                  class="ant-table-body"
                  style="overflow-x: auto; overflow-y: hidden;"
                >
                  <table
                    style="width: auto; min-width: 100%; table-layout: fixed;"
                  >
                    <colgroup>
                      <col
                        style="width: 145px;"
                      />
                      <col
                        style="width: 145px;"
                      />
                    </colgroup>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              demo-app
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              4.35 s
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              customer
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              431 ms
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              mysql
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              431 ms
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              frontend
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              287 ms
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              driver
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              230 ms
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              route
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              66.4 ms
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              redis
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            <div
                              class="line-clamped-text"
                            >
                              31.3 ms
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
